tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -113,164 +111,116 @@ internal ExporterSettings(IConfigurationSource? source, Func<string, bool> fileE
+        public bool PartialFlushEnabled { get; }
 
         /// <summary>
-        /// Gets or sets the minimum number of closed spans in a trace before it's partially flushed
+        /// Gets the minimum number of closed spans in a trace before it's partially flushed
         /// </summary>
-        [PublicApi]
-        public int PartialFlushMinSpans
-        {
-            get
-            {
-                TelemetryFactory.Metrics.Record(PublicApiUsage.ExporterSettings_PartialFlushMinSpans_Get);
-                return PartialFlushMinSpansInternal;
-            }
-
-            set
-            {
-                TelemetryFactory.Metrics.Record(PublicApiUsage.ExporterSettings_PartialFlushMinSpans_Set);
-                PartialFlushMinSpansInternal = value;
-            }
-        }
-
-        internal int PartialFlushMinSpansInternal
-        {
-            get => _partialFlushMinSpans;
-
-            private set
-            {
-                if (value <= 0)
-                {
-                    throw new ArgumentException("The value must be strictly greater than 0", nameof(PartialFlushMinSpansInternal));
-                }
-
-                // Not recording the error condition above, because it can never actually be used
-                // If we instead rejected `value` and used a default, then we _would_ record it here
-                _telemetry.Record(ConfigurationKeys.PartialFlushMinSpans, value, ConfigurationOrigins.Code);
-                _partialFlushMinSpans = value;
-            }
-        }
+        public int PartialFlushMinSpans { get; }
 
         /// <summary>
         /// Gets the transport used to send traces to the Agent.
         /// </summary>
-        internal TracesTransportType TracesTransport { get; private set; }
+        internal TracesTransportType TracesTransport { get; }
 
         /// <summary>
         /// Gets the transport used to connect to the DogStatsD.
         /// Default is <c>TransportStrategy.Tcp</c>.
         /// </summary>
-        internal MetricsTransportType MetricsTransport { get; private set; }
+        internal MetricsTransportType MetricsTransport { get; }
 
         /// <summary>
-        /// Gets or sets the agent host to use when <see cref="MetricsTransport"/> is <see cref="TransportType.UDP"/>
+        /// Gets the agent host to use when <see cref="MetricsTransport"/> is <see cref="TransportType.UDP"/>
         /// </summary>
-        internal string MetricsHostname { get; private set; }
+        internal string MetricsHostname { get; }
 
         internal List<string> ValidationWarnings { get; }
 
         internal IConfigurationTelemetry Telemetry => _telemetry;
 
+        // internal for testing
+        internal static ExporterSettings Create(Dictionary<string, object?> settings)
+            => new(new DictionaryConfigurationSource(settings.ToDictionary(x => x.Key, x => x.Value?.ToString()!)), new ConfigurationTelemetry());
```

Comment:
`x.Value?.ToString()!`
:stare:

Reply:
Yeah, the trouble is the `DictionaryConfigurationSource` needs a `<string, string>` sooooo 😅 We should merge `DictionaryConfigurationSource` and `DictionaryObjectConfigurationSource` tbh, but separate PR...

Given this is only for testing anyway, meh 😉